respy/tests/random_model.py

```diff
@@ -109,12 +114,25 @@ def generate_random_model(
         n_types = np.random.randint(1, bound_constr["max_types"] + 1)
     if n_type_covariates is None:
         n_type_covariates = np.random.randint(2, 4)
+    if observables is None:
+        n_obs = np.random.randint(0, 3)
+        if n_obs == 0:
+            observables = False
+        else:
+            observables = np.random.randint(1, 4, size=n_obs)
 
     params = csv_template(
-        n_types=n_types, n_type_covariates=n_type_covariates, initialize_coeffs=False
+        n_types=n_types,
+        n_type_covariates=n_type_covariates,
+        observables=observables,
+        initialize_coeffs=False,
     )
     params["value"] = np.random.uniform(low=-0.05, high=0.05, size=len(params))
 
+    if observables is not False:
```

Comment:
Observables are spread across the function. Try to collect all logical pieces after lagged choices are created, because that is where observables are also in the state space.